app/screens/settings-screen/account-screen.tsx

```diff
@@ -192,22 +296,54 @@ export const AccountScreen = ({ navigation }: Props) => {
     }
   }
 
-  let identitySettingTitle: LocalizedString
-  let identitySettingAction: (() => void) | undefined
-  switch (currentLevel) {
-    case AccountLevel.NonAuth:
-      identitySettingTitle = LL.GetStartedScreen.logInCreateAccount()
-      identitySettingAction = () => navigation.navigate("getStarted")
-      break
-    case AccountLevel.Zero:
-      identitySettingTitle = LL.common.backupAccount()
-      identitySettingAction = openUpgradeAccountModal
-      break
-    default:
-      identitySettingTitle = LL.common.phoneNumber()
-      break
+  const tryConfirmEmailAgain = async (email: string) => {
+    try {
+      await deleteEmail()
+
+      const { data } = await setEmailMutation({
+        variables: { input: { email } },
+      })
+
+      const errors = data?.userEmailRegistrationInitiate.errors
+      if (errors && errors.length > 0) {
+        Alert.alert(errors[0].message)
+      }
+
+      const emailRegistrationId = data?.userEmailRegistrationInitiate.emailRegistrationId
+
+      if (emailRegistrationId) {
+        navigation.navigate("emailSetValidation", {
+          emailRegistrationId,
+          email,
+        })
+      } else {
+        console.warn("no flow returned")
+      }
+    } catch (err) {
+      console.error(err, "error in setEmailMutation")
+    } finally {
+      // setLoading(false)
+    }
+  }
+
+  const emailSet = async () => {
+    if (email) {
+      Alert.alert(
```

Comment:
Is it necessary to alert them? Why not just do the action on their behalf?

Reply:
it's only if the email is unverified, it's somewhat of a edge case. 

the reason is that if the email is unverified, there is a warning icon, and so it's to indicate that there is a something that needs to be done, and provide some context around instead of just changing screen. 

also, because there are mutation executed after a tap on this screen (to delete and recreated the email, as right now there is no mutation "try send code again", it takes a second or 2 to get to the screen), it's better if the user know that an action will be expected when tapping the button. a waiting wheel could be handy here but I'm not sure where to put it yet.